src/handler.ts

```diff
@@ -73,14 +80,58 @@ const jumpTable: JumpTable = {
     },
   },
   schema: {
-    Create: async (resourceId: string) => {
-      return format("create schema if not exists %I", resourceId)
+    Create: async (resourceId: string, props: SchemaProps) => {
+      const sql: string[] = [format("create schema if not exists %I", resourceId)]
+      if (props.RoleName) {
+        sql.push(format("GRANT USAGE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(format("GRANT CREATE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(
+          format("GRANT ALL ON ALL TABLES IN SCHEMA %I TO %I", resourceId, props.RoleName)
+        )
+      }
+      return sql
     },
-    Update: async (resourceId: string, oldResourceId: string) => {
-      return format("alter schema %I rename to %I", oldResourceId, resourceId)
+    Update: async (resourceId: string, oldResourceId: string, props: SchemaProps) => {
+      const sql: string[] = []
+      // TODO: revoke old role-name if props.RoleName was removed or changed
```

Comment:
Yeah, that needs doing (and be tested).

Reply:
You did those notes as well:

https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L118
https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L160

If the handler would be prepared to get old props as well I would have done this check. It is out of the scope of this little change to refactor the handler.